src/odin/cli.rs

```diff
@@ -0,0 +1,130 @@
+use clap::{Parser, Subcommand};
+
+#[derive(Parser)]
+#[command(author, version)]
+#[command(propagate_version = true)]
+pub struct Cli {
```

Comment:
I would recommend adding

```rust
#[cfg(test)]
mod tests {
  use super::*;

  use clap::CommandFactory;

  #[test]
  fn asserts() {
    Cli::command().debug_assert();
  }
}
```

to the bottom of the file. `clap` gave up trying to enforce _everything_ at compile time, so they added `.debug_assert()` as a way to automatically test for some issues. This picked up on the conflicting names with some of the short args in `configure`